src/cudamatrix/cu-kernels.cu

```diff
@@ -2336,7 +2357,14 @@ void cudaF_replace_value(int Gr, int Bl, float *v, int dim, float orig, float ch
 void cudaF_set_bias_params(int Gr, int Bl, float* v, const float* a, float param_1, float param_2, float param_3, int* flag, int dim) {
   _set_bias_params<<<Gr,Bl>>>(v,a,param_1,param_2,param_3,flag,dim);
 }
-
+void cublas_copy_kaldi_fd(int Gr, int Bl, int n, const float* x, 
+		int incx, double* y, int incy){
+  _copy_from_vec<<<Gr,Bl>>>(n,x,incx,y,incy);
+}
+void cublas_copy_kaldi_df(int Gr, int Bl, int n, const double* x, 
+		int incx, float* y, int incy){
+  _copy_from_vec<<<Gr,Bl>>>(n,x,incx,y,incy);
+}
 void cudaF_copy_from_vec_df(int Gr, int Bl, double* v_out, const float* v_in, int dim) {
   _copy_from_vec_df<<<Gr,Bl>>>(v_out,v_in,dim);
```

Comment:
I don't see shy we need _copy_from_vec_df and so on... it's all templated on the types.